cpu/stm32_common/include/pindbg_cpu.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_stm32_common
+ *
+ * @{
+ * @file
+ * @brief       CPU specific handlers for direct control of debug pins
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef PINDBG_CPU_H
+#define PINDBG_CPU_H
+
+#include "cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef PINDBG0
```

Comment:
IMHO it would be great to make this more generic, that is for arbitrary numbers of debug pins, and make the config consistent with other peripheral confs. So I was thinking about something like:

```
static const pindbg_cfg_t pindbg_config[] = {
    GPIO_PIN(0, 1),
    GPIO_PIN(0, 2),
}

#define PINDGB_NUMOF ( sizeof(pindbg_config) / sizeof(pindbg_config[0] )
```

and then also have MCU specific macros like

```
#define PINDBG_PORT(x)        ((GPIO_TypeDef *)(x & ~(0x0f)))
#define PINDBG_MASK(x)        (1 << (x & 0xf))
#define PINDBG_SET(x)         (PINDBG_PORT(x)->BSRR = PINDBG_MASK(x))
#define PINDBG_CLR(x)         (PINDBG_PORT(x)->BSRR = (PINDBG_MASK(x) << 16))
#define PINDBG_TGL(x)         (PINDBG_PORT(x)->ODR ^= PINDBG_MASK(x))
```

which then would turn `pindbg_init()` into a simple and generic for-loop.

@haukepetersen you think that's possible and handy?

Reply:
With this I see two problems:
- it is not possible to simply redefine a pin during compile time, e.g. `DBGPIN0=GPIO_PIN\(0,2\) make ...`
- I am not sure, if the set/clr/tgl expressions will still evaluate to a single instruction as they do in the current code

Reply:
for the first:

```
#ifndef PINDBG0
#define PINDBG0 GPIO_PIN(0, 1)
#endif
#ifndef PINGDBG1
#define PINDBG0 GPIO_PIN(0, 2)
#endif 

static const pindbg_cfg_t pindbg_config[] = {
    PINDBG0,
    PINDBG1,
}
```

and for the second: its preprocessor text replacement, i.e, my example above should resolve to the same statement as you static defines where ever it is used in the code.